### PR TITLE
power: Add OS Power Management debug hooks

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
Add the necessary PM hooks for logging and debugging
purposes.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

### CODE REVIEW

File: subsys/power/power.c [hunk 1]
```diff
@@ -18,6 +19,78 @@ LOG_MODULE_REGISTER(power);
...
+		timer_end = rtc_read(pm_timer);
+	}
+#endif
+}
+
+static void sys_pm_debug_setup_timers(void)
+{
+
+#ifdef CONFIG_RTC
+	pm_timer = device_get_binding(CONFIG_PM_DEBUG_RTC_DEV_NAME);
```
Comment (pizi-nordic) [inline subsys/power/power.c]: Do we need dedicated RTC for that? Why we cannot use ticks or system timer cycles?

File: subsys/power/power.c [hunk 2]
```diff
@@ -18,6 +19,78 @@ LOG_MODULE_REGISTER(power);
 static int post_ops_done = 1;
 static enum power_states pm_state;
 
+#ifdef CONFIG_PM_CONTROL_OS_DEBUG
+
+struct pm_debug_info {
+	u32_t count;
+	u32_t last_res;
```
Comment (pizi-nordic) [inline subsys/power/power.c]: IMHO average will give more information than last.

File: subsys/power/power.c [hunk 3]
```diff
@@ -100,6 +180,9 @@ static int sys_pm_init(struct device *dev)
 	ARG_UNUSED(dev);
 
 	sys_pm_create_device_list();
+
```
Comment (pizi-nordic) [inline subsys/power/power.c]: Empty line not needed.

**Review by pizi-nordic — COMMENTED** [review]
Could we register a CLI command to fetch statistics of the power management?

File: subsys/power/power.c (hunk 1 above)
Comment (nashif) [reply subsys/power/power.c]: agree. Not sure why we are using RTC here, not many platforms have this. Seems to be legacy from when power management was focused on devices with RTC driver (!= system timer)

**Comment by nashif** [comment]
no need for subsys: in the commit message.

**Comment by ramakrishnapallala** [comment]
@pizi-nordic @nashif  I have addressed the comments and updated the PR. Please have a look.

**Review by pizi-nordic — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**